### ISSUE TITLE
Vite config option to build esm for an ssr build

### ISSUE THREAD

**Issue by adamdbradley** [BuilderIO/qwik#470]
### Is your feature request related to a problem?

Netlify edge uses ESM, but vite's ssr default is CJS.

### Describe the solution you'd like

Either add a cli flag, or vite config option, or both.

### Describe alternatives you've considered

https://github.com/netlify/vite-plugin-netlify-edge/blob/main/src/index.ts

**Comment by kylecordes** [comment]
Since Qwik is pretty cutting edge at this point, could it be plausibly reasonable to make ESM the default output everywhere? CJS as a fallback for compatibility with legacy hosting.

**Comment by ascorbic** [comment]
Good point. Cloudflare Workers also support ESM and the qwik `engines` field requires Node >=14, so I'm not sure which environments would _not_ support it.

**Comment by adamdbradley** [comment]
Since there's already a setting in Vite to set the target to `webworker` I think we'll just document to use that instead. https://vitejs.dev/guide/ssr.html#ssr-specific-plugin-logic
